app/src/components/TransferArtifact.tsx

```diff
@@ -97,29 +98,37 @@ const TransferArtifact: React.FC<TransferArtifactProps> = ({ tokenId, metaUri })
       fields.date,
     );
 
-    const takesArr = ARR_LOCATIONS.includes(fields.location);
-
-    ArtifactRegistry.methods.transfer(
-      owner,
-      recipientAddress,
-      tokenId,
-      provenanceHash,
-      (parseFloat(fields.price) * 100).toString(),
-      fields.location,
-      fields.date,
-      takesArr,
-    ).send(
-      {
-        from: accounts[0],
-        gasLimit: 6000000,
-      },
-    ).then(() => {
-      setSubmitted(false);
-    }).catch((err: any) => {
-      // rejection, usually
-      console.log(err);
-      setSubmitted(false);
-    });
+    const eventOptions = { fromBlock: 0 };
+
+    ArtifactRegistry.getPastEvents('RecordSale', eventOptions)
+      .then((events: EventData[]) => {
+        return events.filter(event => event.returnValues.tokenId === tokenId.toString());
+      }).then((relevantEvents: EventData[]) => {
+        // only take ARR in country that takes it, or if no sales with this token have occurred
+        // no sales with this token → current user is the one who registered it → they're the artist
```

Comment:
Either the current user is the artist, or a gallery who registered it on their behalf 
(in both cases can ignore ARR)